addons/binding/org.openhab.binding.pentair/src/main/java/org/openhab/binding/pentair/handler/PentairSerialBridgeHandler.java

```diff
@@ -0,0 +1,215 @@
+
+            if (cp instanceof SerialPort) {
+                port = (SerialPort) cp;
+            } else {
+                throw new IllegalStateException("unknown port type");
+            }
+            port.setSerialPortParams(9600, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+            port.disableReceiveFraming();
+            port.disableReceiveThreshold();
+
+            reader = new BufferedInputStream(port.getInputStream());
+            writer = new BufferedOutputStream(port.getOutputStream());
+            logger.info("Pentair Bridge connected to serial port: {}", configuration.serialPort);
+        } catch (PortInUseException e) {
+            String msg = String.format("cannot open serial port: %s", configuration.serialPort);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+            return;
+        } catch (UnsupportedCommOperationException e) {
+            String msg = String.format("got unsupported operation %s on port %s", e.getMessage(),
+                    configuration.serialPort);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+            return;
+        } catch (NoSuchPortException e) {
+            String msg = String.format("got no such port for %s", configuration.serialPort);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+            return;
+        } catch (IllegalStateException e) {
+            String msg = String.format("got unknown port type for %s", configuration.serialPort);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+            return;
+        } catch (IOException e) {
+            String msg = String.format("IOException on port %s", configuration.serialPort);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, msg);
+            return;
+        }
+
+        parser = new Parser();
+        thread = new Thread(parser);
+        thread.start();
+
+        if (port != null && reader != null && writer != null) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Unable to connect");
+        }
+    }
+
+    @Override
+    protected synchronized void disconnect() {
+        updateStatus(ThingStatus.OFFLINE);
+
+        if (thread != null) {
+            try {
+                thread.interrupt();
+                thread.join(); // wait for thread to complete
+            } catch (InterruptedException e) {
+                // do nothing
+            }
+            thread = null;
+            parser = null;
+        }
+
+        if (reader != null) {
+            try {
+                reader.close();
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Error in closing reader");
```

Comment:
We were already closing down, you can ignore or trace/debug log the exception but don't have to change the thing status. 2x